dipy/viz/horizon/tab/slice.py

```diff
@@ -3,396 +3,378 @@
-        self.__tab_ui.add_element(
-            self.__tab_id, self.__slider_slice_y, (x_pos, .38))
-        self.__tab_ui.add_element(
-            self.__tab_id, self.__slider_slice_z, (x_pos, .15))
+        x_pos = .02
+        self._slice_x_toggle.position = (x_pos, .62)
+        self._slice_y_toggle.position = (x_pos, .38)
+        self._slice_z_toggle.position = (x_pos, .15)
+
+        x_pos = .05
+        self._slice_opacity.label.position = (x_pos, .85)
+        self._slice_x.label.position = (x_pos, .62)
+        self._slice_y.label.position = (x_pos, .38)
+        self._slice_z.label.position = (x_pos, .15)
+
+        x_pos = .10
+        self._slice_opacity.element.position = (x_pos, .85)
+        self._slice_x.element.position = (x_pos, .62)
+        self._slice_y.element.position = (x_pos, .38)
+        self._slice_z.element.position = (x_pos, .15)
 
         x_pos = .52
-
-        self.__tab_ui.add_element(
-            self.__tab_id, self.__slider_label_intensities, (x_pos, .85))
-        self.__tab_ui.add_element(
-            self.__tab_id, self.__buttons_label_colormap, (x_pos, .56))
-        self.__tab_ui.add_element(
-            self.__tab_id, self.__picker_label_voxel, (x_pos, .38))
+        self._intensities.label.position = (x_pos, .85)
+        self._colormap_switcher.label.position = (x_pos, .56)
+        self._voxel_label.position = (x_pos, .38)
 
         x_pos = .60
+        self._intensities.element.position = (x_pos, .85)
+        self._colormap_switcher.element.position = [
+            (x_pos, .54), (0.63, .54), (0.69, .54)
+        ]
+        self._voxel_data.position = (x_pos, .38)
 
-        self.__tab_ui.add_element(
-            self.__tab_id, self.__slider_intensities, (x_pos, .85))
-        self.__tab_ui.add_element(
-            self.__tab_id, self.__button_previous_colormap, (x_pos, .54))
-        self.__tab_ui.add_element(
-            self.__tab_id, self.__label_selected_colormap, (.63, .56))
-        self.__tab_ui.add_element(
-            self.__tab_id, self.__button_next_colormap, (.69, .54))
-        self.__tab_ui.add_element(
-            self.__tab_id, self.__label_picked_voxel, (x_pos, .38))
-
-        data_ndim = len(self.__data_shape)
-
-        if data_ndim == 4:
+        if len(self._visualizer.data_shape) == 4:
             x_pos = .52
-            self.__tab_ui.add_element(
-                self.__tab_id, self.__slider_label_volume, (x_pos, .15))
+            self._volume.label.position(x_pos, .15)
 
             x_pos = .60
-            self.__tab_ui.add_element(
-                self.__tab_id, self.__slider_volume, (x_pos, .15))
+            self._volume.element.position = (x_pos, .15)
 
     @property
     def name(self):
-        return self.__name
+        return self._name
+
+    @property
+    def tab_id(self):
+        """
+        Id of the tab.
+        """
+        return self._tab_id
+
+    @property
+    def tab_type(self):
+        return 'slices_tab'
```

Comment:
to remove

Reply:
We cannot remove the name as that becomes title of the tab by tabmanager